image_publisher/src/nodelet/image_publisher_nodelet.cpp

```diff
@@ -0,0 +1,162 @@
+
+#include <ros/ros.h>
+#include <nodelet/nodelet.h>
+#include <cv_bridge/cv_bridge.h>
+#include <image_publisher/ImagePublisherConfig.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <camera_info_manager/camera_info_manager.h>
+#include <opencv2/highgui/highgui.hpp>
+#include <dynamic_reconfigure/server.h>
+#include <boost/assign.hpp>
+using namespace boost::assign;
+
+namespace image_publisher {
+class ImagePublisherNodelet : public nodelet::Nodelet
+{
+  dynamic_reconfigure::Server<image_publisher::ImagePublisherConfig> srv;
+
+  image_transport::CameraPublisher pub_;
+
+  boost::shared_ptr<image_transport::ImageTransport> it_;
+  ros::NodeHandle nh_;
+
+  cv::VideoCapture cap_;
+  cv::Mat image_;
+  int subscriber_count_;
+  ros::Timer timer_;
+
+  std::string frame_id_;
+  std::string filename_;
+  sensor_msgs::CameraInfo camera_info_;
+  
+
+  void reconfigureCallback(image_publisher::ImagePublisherConfig &new_config, uint32_t level)
+  {
+    frame_id_ = new_config.frame_id;
+
+    timer_ = nh_.createTimer(ros::Duration(1.0/new_config.publish_rate), &ImagePublisherNodelet::do_work, this);
+
+    camera_info_manager::CameraInfoManager c(nh_);
+    if ( !new_config.camera_info_url.empty() ) {
+      try {
+        c.validateURL(new_config.camera_info_url);
+        c.loadCameraInfo(new_config.camera_info_url);
+        camera_info_ = c.getCameraInfo();
+      } catch(cv::Exception &e) {
+        NODELET_ERROR("camera calibration failed to load: %s %s %s %i", e.err.c_str(), e.func.c_str(), e.file.c_str(), e.line);
+      }
+    }
+    
+
+  }
+
+  void do_work(const ros::TimerEvent& event)
+  {
+    // Transform the image.
+    try
+    {
+      if ( cap_.isOpened() ) {
+        if ( ! cap_.read(image_) ) {
+          cap_.set(CV_CAP_PROP_POS_FRAMES, 0);
+        }
+      }
+      sensor_msgs::ImagePtr out_img = cv_bridge::CvImage(std_msgs::Header(), "bgr8", image_).toImageMsg();
+      out_img->header.frame_id = frame_id_;
+      out_img->header.stamp = ros::Time::now();
+      camera_info_.header.frame_id = out_img->header.frame_id;
+      camera_info_.header.stamp = out_img->header.stamp;
+
+      pub_.publish(*out_img, camera_info_);
+    }
+    catch (cv::Exception &e)
+    {
+      NODELET_ERROR("Image processing error: %s %s %s %i", e.err.c_str(), e.func.c_str(), e.file.c_str(), e.line);
+    }
+  }
+
+  void connectCb(const image_transport::SingleSubscriberPublisher& ssp)
+  {
+    subscriber_count_++;
+  }
+
+  void disconnectCb(const image_transport::SingleSubscriberPublisher&)
+  {
+    subscriber_count_--;
+  }
+
+public:
+  virtual void onInit()
+  {
+    subscriber_count_ = 0;
+    nh_ = getNodeHandle();
+    it_ = boost::shared_ptr<image_transport::ImageTransport>(new image_transport::ImageTransport(nh_));
+    pub_ = image_transport::ImageTransport(nh_).advertiseCamera("image_raw", 1);
+
+    nh_.param("filename", filename_, std::string(""));
```

Comment:
Why not see `~filename`? I usually run multiple image_publisher nodes without using namespace.